src/main/java/com/mixpanel/android/mpmetrics/AnalyticsMessages.java

```diff
@@ -298,7 +298,7 @@ else if (msg.what == KILL_WORKER) {
                     ///////////////////////////
 
                     if ((returnCode >= mConfig.getBulkUploadLimit() || returnCode == MPDbAdapter.DB_OUT_OF_MEMORY_ERROR) &&
-                            SystemClock.elapsedRealtime() >= mRetryAfter) {
+                            SystemClock.elapsedRealtime() >= mRetryAfter && mFailedRetries <= 0) {
```

Comment:
The problem with that is that we would not schedule any job. That means that we'd have to rely on the user to track a new event, for example, and when that happen is when we check for `SystemClock.elapsedRealtime()` and compare with a minimum threshold (`mRetryAfter`). Implementation wise I believe it makes more sense to schedule a task in the future than rely on the activity on the user for it to be executed.